### ISSUE TITLE
DDC bug on await/cascade in a list

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/sdk#28338]
The following is failing on DDC:

```dart
import 'dart:async';
 
class Bar {
  int x = 42;
  int y;

  String toString() => "($x,$y)";
}

class Foo {

  Future<Bar> a() async => new Bar();

  Future<List> b() async {
    var result = [(await a())..y = 38];
    return result;
  }
}

main() async {
  var foo = new Foo();
  var result = await foo.b();
  print(result);
}
```

with this runtime error:

```
TypeError: Cannot read property 'a' of undefined
```

It appears that `this` is unbound in `Foo.b`.

**Comment by vsmenon** [comment]
Related to the recently fixed #28032

**Comment by vsmenon** [comment]
(updated example in first comment)
